DoSafely() doesn't actually do anything

When running a task with `DoSafely`, it never actually performs the task.

Here is an example main.go file:

```
package main

import (
	"io/ioutil"
	"log"

	"github.com/jasonlvhit/gocron"
)

func task() {
	log.Println("I am runnning task.")
	d1 := []byte("hello\ngo\n")
	ioutil.WriteFile("/tmp/dat1", d1, 0644)
}

func main() {
	s := gocron.NewScheduler()
	s.Every(3).Seconds().DoSafely(task)
	<-s.Start()
}
```

Test it out - if you swap in `Do` for `DoSafely` it works as expected, outputs to the terminal and writes the file. With `DoSafely` nothing happens.

If I run a test just for the purpose of debugging, it generates a panic
```
func TestDoSafely(t *testing.T) {
	success := false
	Every(1).Second().DoSafely(mutatingTask, &success)

	<-Start()
}
```

```
=== RUN   TestDoSafely
panic: reflect: Call using []interface {} as type *bool

goroutine 21 [running]:
reflect.Value.call(0x133fae0, 0x13d4a70, 0x13, 0x13be056, 0x4, 0xc000144000, 0x1, 0x1, 0x5dbc7f46, 0x7ab5548, ...)
	/usr/local/Cellar/go/1.12.9/libexec/src/reflect/value.go:377 +0x1457
reflect.Value.Call(0x133fae0, 0x13d4a70, 0x13, 0xc000144000, 0x1, 0x1, 0x0, 0x0, 0x0)
	/usr/local/Cellar/go/1.12.9/libexec/src/reflect/value.go:308 +0xa4
github.com/jasonlvhit/gocron.(*Job).run(0xc0000cadc0, 0x0, 0x0, 0x0, 0x0, 0x0)
	/Users/john/go/src/github.com/jasonlvhit/gocron/gocron.go:129 +0x415
github.com/jasonlvhit/gocron.(*Scheduler).RunPending(0xc000118000)
	/Users/john/go/src/github.com/jasonlvhit/gocron/gocron.go:516 +0xa0
github.com/jasonlvhit/gocron.(*Scheduler).Start.func1(0xc0000cc410, 0xc000118000, 0xc00010e1c0)
	/Users/john/go/src/github.com/jasonlvhit/gocron/gocron.go:605 +0x42
created by github.com/jasonlvhit/gocron.(*Scheduler).Start
	/Users/john/go/src/github.com/jasonlvhit/gocron/gocron.go:601 +0x87

Process finished with exit code 1
```

--- a/gocron.go
+++ b/gocron.go
@@ -115,8 +115,8 @@ func (j *Job) run() (result []reflect.Value, err error) {
 		}()
 	}
 
-	result, err = callJobFuncWithParams(j.funcs[j.jobFunc], j.fparams[j.jobFunc])
 	j.lastRun = time.Now()
+	result, err = callJobFuncWithParams(j.funcs[j.jobFunc], j.fparams[j.jobFunc])
 	j.scheduleNextRun()
 	return
 }
@@ -166,10 +166,7 @@ func (j *Job) DoSafely(jobFun interface{}, params ...interface{}) {
 			}
 		}()
 
-		_, err := callJobFuncWithParams(jobFun, params)
-		if err != nil {
-			log.Printf("ERROR: %v", err)
-		}
+		_, _ = callJobFuncWithParams(jobFun, params)
 	}
 
 	j.Do(recoveryWrapperFunc, params...)

--- a/gocron.go
+++ b/gocron.go
@@ -115,20 +115,22 @@ func (j *Job) run() (result []reflect.Value, err error) {
 		}()
 	}
 
-	f := reflect.ValueOf(j.funcs[j.jobFunc])
-	params := j.fparams[j.jobFunc]
+	result, err = callJobFuncWithParams(j.funcs[j.jobFunc], j.fparams[j.jobFunc])
+	j.lastRun = time.Now()
+	j.scheduleNextRun()
+	return
+}
+
+func callJobFuncWithParams(jobFunc interface{}, params []interface{}) ([]reflect.Value, error) {
+	f := reflect.ValueOf(jobFunc)
 	if len(params) != f.Type().NumIn() {
-		err = errors.New("the number of param is not adapted")
-		return
+		return nil, errors.New("the number of params is not matched")
 	}
 	in := make([]reflect.Value, len(params))
 	for k, param := range params {
 		in[k] = reflect.ValueOf(param)
 	}
-	j.lastRun = time.Now()
-	result = f.Call(in)
-	j.scheduleNextRun()
-	return
+	return f.Call(in), nil
 }
 
 // for given function fn, get the name of function.
@@ -157,13 +159,20 @@ func (j *Job) Do(jobFun interface{}, params ...interface{}) {
 
 // DoSafely does the same thing as Do, but logs unexpected panics, instead of unwinding them up the chain
 func (j *Job) DoSafely(jobFun interface{}, params ...interface{}) {
-	defer func() {
-		if err := recover(); err != nil {
-			log.Printf("Internal panic occurred: %s", err)
-		}
-	}()
+	recoveryWrapperFunc := func() {
+		defer func() {
+			if r := recover(); r != nil {
+				log.Printf("Internal panic occurred: %s", r)
+			}
+		}()
 
-	j.Do(jobFun, params)
+		_, err := callJobFuncWithParams(jobFun, params)
+		if err != nil {
+			log.Printf("ERROR: %v", err)
+		}
+	}
+
+	j.Do(recoveryWrapperFunc, params...)
 }
 
 func Jobs() []*Job {

--- a/gocron_test.go
+++ b/gocron_test.go
@@ -18,14 +18,6 @@ func taskWithParams(a int, b string) {
 	fmt.Println(a, b)
 }
 
-func mutatingTask(mutableValue *bool) {
-	*mutableValue = !*mutableValue
-}
-
-func failingTask() {
-	log.Panic("I am panicking!")
-}
-
 func TestSecond(t *testing.T) {
 	sched := NewScheduler()
 	job := sched.Every(1).Second()
@@ -65,20 +57,24 @@ func testJobWithInterval(t *testing.T, sched *Scheduler, job *Job, expectedTimeB
 func TestSafeExecution(t *testing.T) {
 	sched := NewScheduler()
 	success := false
-	sched.Every(1).Second().Do(mutatingTask, &success)
+	sched.Every(1).Second().Do(func(mutableValue *bool) {
+		*mutableValue = !*mutableValue
+	}, &success)
 	sched.RunAll()
 	assert.Equal(t, true, success, "Task did not get called")
 }
 
 func TestSafeExecutionWithPanic(t *testing.T) {
 	defer func() {
-		if err := recover(); err != nil {
-			t.Errorf("Unexpected internal panic occurred: %s", err)
+		if r := recover(); r != nil {
+			t.Errorf("Unexpected internal panic occurred: %s", r)
 		}
 	}()
 
 	sched := NewScheduler()
-	sched.Every(1).Second().DoSafely(failingTask)
+	sched.Every(1).Second().DoSafely(func() {
+		log.Panic("I am panicking!")
+	})
 	sched.RunAll()
 }
 

